platform/linux-generic/odp_queue_lf.c

```diff
@@ -24,14 +24,55 @@
 
 typedef unsigned __int128 u128_t;
 
+static inline void atomic_zero_u128(u128_t *atomic)
+{
+	__atomic_store_n(atomic, 0, __ATOMIC_RELAXED);
+}
+
+#if defined(__aarch64__)
+/* ARMv8 has atomic load-acq/store-rel instructions for a pair of
+ * 64bit of data. GCC atomic built-in for 128bits does not utilize these
+ * instructions but uses locks instead. Override GCC built-in for ARMv8.
+ */
```

Comment:
That sounds like not complete gcc support. Which compiler/version did you use? Isn't arch/aarch64/odp_atomic.h  more suitable for this?

Reply:
At least GCC 5.4, 7.1 and 8.1 does not support lock-free 16 byte operations.

In it's current form odp_atomic.h cannot be included directly, but through odp_cpu.h:

#ifndef PLATFORM_LINUXGENERIC_ARCH_ARM_ODP_CPU_H
#error This file should not be included directly, please include odp_cpu.h
#endif